port/network_mbedtls_wrapper.c

```
@@ -154,6 +159,20 @@ IoT_Error_t iot_tls_connect(Network *pNetwork, TLSConnectParams *params) {
     ESP_LOGD(TAG, "ok (%d skipped)", ret);
 
     /* Load client certificate... */
+#ifdef CONFIG_AWS_IOT_USE_HARDWARE_SECURE_ELEMENT
+    if (pNetwork->tlsConnectParams.pDeviceCertLocation[0] == '#') {
+        int ret;
+        const atcacert_def_t* cert_def = NULL;
+
+        ESP_LOGD(TAG, "Using certificate stored in ATECC608a");
+        ret = tng_get_device_cert_def(&cert_def);
+        if (ret == ATCACERT_E_SUCCESS) {
+            ret = atca_mbedtls_cert_add(&(tlsDataParams->clicert), cert_def);
+        } else {
+            ESP_LOGE(TAG, "Failed to load cert from ATECC608a, tng_get_device_cert_def returned %02x", ret);
```

This should return an appropriate error code